ansible/roles/email-server-config/tasks/main.yml

```diff
@@ -0,0 +1,84 @@
+---
+- name: Install required packages
+  package:
+    name: '{{ item }}'
+    state: present
+  with_items: '{{ required_packages }}'
+
+- name: Remove any exisitng postfix configs
+  file:
+    path: "/etc/postfix/{{ item }}"
+    state: absent
+  with_items: '{{ postfix_config }}'
+
+- name: Install postfix config configs
+  copy:
+    src: "{{ item }}"
+    dest: "/etc/postfix/{{ item }}"
+    mode: 0644
+  with_items: '{{ postfix_config }}' 
+
+- name: Remove any exisitng opendkim configs
+  file:
+    path: "{{ opendkim_config_path }}"
+    state: absent
+
+- name: Install opendkim.conf
+  copy:
+    src: "{{ opendkim_conf }}"
+    dest: "{{ opendkim_config_path }}"
+    mode: 0644
+
+- name: Create required directories
+  file:
+    path: "{{ item[0] }}"
+    state: directory
+    owner: '{{ item[1] }}'
+    group: '{{ item[2] }}'
+    mode: 0750
+  loop: "{{ required_dirs|product(users, user_groups)|list }}"
```

Comment:
Is the opendkim user responsible for the spool/filter directory?

Reply:
This should probably be root, I have changed the code so the directory will be created as root ownership.